### PR TITLE
feat(model): extend fit to SIR-X 

### PR DESCRIPTION
**Pull request by felipehuerta17**
This pull request enhances the model.fit() function in terms of its
accuracy and robustness:

In model.py, an special case for optimization is included for
the SIR-X model. One of the model parameters is the initial ratio
between infected and tested individuals, i_0/x_0. This is fitted
and require a slightly more careful treatment than traditional
parameter fitting. You can see the implementation on model.py fit

As I discovered this fifth parameter of sirx, I modified the sirx.py
class increasing the number of parameters to 5 and unpacking them
properly. Additionally, I included interesting quantities mentioned
in the paper of Maier et al (2020) such as the public containment
leverage P and the probability of quarantined Q. An effective r_0
and T_I are also defined. All of this was implemented as properties
in sirx.

I included two more attributes on SIR and SIR-X:
- name: this enables the fit() function to adjust the procedure
for the SIR or SIR-X (and will be helpful to extend for any other model)
- fit_input: Although SIR fits "I" directly from the number of
observed cases, SIR-X fits "X" to the observed cases and calculates I.
To specify to which index of the model outputs the input data should
be fitted, the fit_input is a property that the user can control.
By default, fit_input = 2 for SIR and fit_input = 4 for SIR-X

### CODE REVIEW

File: opensir/models/sirx.py [hunk 1]
```diff
@@ -37,15 +46,15 @@ class SIRX(Model):
     """ SIRX model definition """
 
     CSV_ROW = ["Days", "S", "I", "R", "X"]
-    NUM_PARAMS = 4
+    NUM_PARAMS = 5
```
Comment (jia200x) [inline opensir/models/sirx.py:49]: why is this 5? are there only 4 parameters? (alpha, betta, kappa_0, kappa)

Comment (felipehuerta17) [reply opensir/models/sirx.py:49]: There is a parameter which needs to be fitted, i_0/x_0. See fit function.

File: opensir/models/model.py [hunk 2]
```diff
@@ -145,12 +147,20 @@ def function_handle(t, *par_fit, pop=population):
             params = np.array(self.p)
             params[fit_index] = par_fit
             self.p = params
-            i_mod = call_solver(self._model, self.p, self.w0, t)
-            return i_mod[:, 2] * pop
+            # SIR-X special fitting of initial conditions
+            if self.name == "sirx":
```
Comment (jia200x) [inline opensir/models/model.py:151]: All children specific code should be in their respective classes, not in the parent class.
This pattern should be modelled with e.g a "get_initial_conds()" function implements by both children classes (SIR and SIR-X)

File: opensir/models/model.py [hunk 3]
```diff
@@ -37,6 +37,8 @@ def __init__(self):
         self.pop = None
         self.w0 = None
         self.pcov = None
+        self.fit_input = None
```
Comment (jia200x) [inline opensir/models/model.py:40]: This is a class dependent constant, not an instance dependent constant. Thus, it should be modeled as the CSV_ROW or NUM_PARAMS parameters

File: opensir/models/sirx.py [hunk 4]
```diff
@@ -16,18 +16,27 @@ def sirx(w, t, p):
         X: Fraction of the population that is quarantined
 
     t: time
-    p: vector of parameter
+    p: vector of parametersÑ
+
+        alpha: transmission rate
```
Comment (jia200x) [inline opensir/models/sirx.py:21]: Did you check if this is displayed properly in Sphinx?
```
pipenv run doc
firefox _build/htm/index.html
```

File: opensir/models/sirx.py (hunk 1 above)
Comment (jia200x) [reply opensir/models/sirx.py:49]: ah! I see it received another parameters

**Comment by jia200x** [comment]
The documentation got broken by #33 . I already fixed it in #36 , so maybe we should merge that one first to see if the documentation displays ok :)

**Comment by felipehuerta17** [comment]
@all-contributors please add @jia200x for code, documentation and maintenance
